### PR TITLE
[Work in progress] new styling

### PR DESCRIPTION
**Pull request by drmenzelit**
### Summary of Changes

- Several changes on scss files
- added color variables
- changed darken() to scale-color()
- change to hsl colors
- correct margins and paddings
- structure changes in index.php to adapt to the new design

### Testing Instructions
Install the PR and run npm ci
Use the theme switcher in the template

### Expected result
You will see the design changing the colors

One issue: the current Cassiopeia logo doesn't change the color yet. If you want to see the color switcher working here, you need to write a title in the template configuration

### CODE REVIEW

File: templates/cassiopeia/scss/blocks/_header.scss [hunk 1]
```diff
@@ -66,14 +59,13 @@
...
-    margin: 0;
     padding: 0;
+    margin: 0;
     list-style: none;
-    color: $white;
 
     @include media-breakpoint-up(md) {
-      flex: 1 1 0%;
       display: flex;
+      flex: 1 1 0%;
```
Comment (brianteeman) [inline templates/cassiopeia/scss/blocks/_header.scss]: The third value is width so when it is zero you dont use the unit or %

```suggestion
      flex: 1 1 0;
```

File: templates/cassiopeia/scss/offline.scss [hunk 2]
```diff
@@ -5,8 +5,8 @@
 @import "tools/tools";
 
 html {
-  background: radial-gradient(circle, $gray-100, $gray-200);
-  background-color: $cyan;
+  background: radial-gradient(circle, gray(gray-100), gray(gray));
+  background-color: color(cyan);
```
Comment (brianteeman) [inline templates/cassiopeia/scss/offline.scss]: I dont understand this syntax but where you used it for white it was wrapped in " quotes

File: templates/cassiopeia/scss/blocks/_header.scss (hunk 1 above)
Comment (drmenzelit) [reply templates/cassiopeia/scss/blocks/_header.scss]: Corrected in last commit

File: templates/cassiopeia/scss/offline.scss (hunk 2 above)
Comment (drmenzelit) [reply templates/cassiopeia/scss/offline.scss]: Syntax corrected in last commit. But the offline page needs some love ;-)

**Comment by richard67** [comment]
When enabled, the back-to-top button still has the colors of the core Cassiopeia, i.e. it doesn't follow the new color themes.

**Comment by richard67** [comment]
With color theme "Winter", buttons like e.g. of the search or the login module have black text on black background. Only if hovered you can see the button text (black text on grey background).

**Comment by richard67** [comment]
@drmenzelit With the winter color scheme, the active menu item is a very light grey text on white background. That is not a11y. When hovering, colors are ok.

**Comment by richard67** [comment]
With spring color theme it is a very light green on white background, also not really good .

**Comment by richard67** [comment]
@drmenzelit If you enter in the template style settings a site description, you will not see it in frontend because it is white text on white background (`<div class="site-description">`).

**Comment by richard67** [comment]
@drmenzelit Question: Is it by purpose that when sticky header is enabled in the template style settings, only the logo and the search module are sticky but the menu is not?

**Comment by drmenzelit** [comment]
> @drmenzelit If you enter in the template style settings a site description, you will not see it in frontend because it is white text on white background (`<div class="site-description">`).

It is corrected now

**Comment by drmenzelit** [comment]
> @drmenzelit Question: Is it by purpose that when sticky header is enabled in the template style settings, only the logo and the search module are sticky but the menu is not?

I missed this as I took the nav outside the header. 
As the header with logo & search in one line and the navigation on other line takes too much place, I decided to make only the navigation sticky.

**Comment by richard67** [comment]
Much better now.

**Comment by richard67** [comment]
The "Prev" and "Next" links for articles still use the core Cassiopeia colour:

snap-1

**Comment by drmenzelit** [comment]
> 
> 
> The "Prev" and "Next" links for articles still use the core Cassiopeia colour:
> 
I'm sure I changed it... it should use the color of the selected theme...

**Comment by richard67** [comment]
Well the change was after my test.

**Comment by drmenzelit** [comment]
> 
> 
> Well the change was after my test.

hahaha... I saw "still" and I thought it is a new comment... I'm tired...

**Comment by richard67** [comment]
As we have discussed in the Team on Glip, we will stop this approach of the new design here for now in favour of PR #71 . I close this PR so that interested people and testers are not confused, but I suggest we don't delete the branch but keep it in case if we later want to come back to it.

**Pull request closed without merging**